### PR TITLE
add easyconfigs for pplacer and OCaml deps (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
@pforai: this is one of the dependencies for QIIME...

depends on ~~https://github.com/hpcugent/easybuild-easyblocks/pull/467~~

### CODE REVIEW

File: easybuild/easyconfigs/o/OCaml/OCaml-4.01.0-ictce-5.5.0.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+toolchain = {'name': 'ictce', 'version': '5.5.0'}
+
+sources = ['ocaml-%(version)s.tar.gz']
+sourcedir = name.lower() + '-' + '.'.join(version.split('.')[0:-1])
+source_urls = ['http://caml.inria.fr/pub/distrib/%s' % sourcedir]
+
+patches = ['OCaml-%(version)s_icc-fixes.patch']
+
+builddependencies = [('Autoconf', '2.69')]
+dependencies = [('ncurses', '5.9')]
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/o/OCaml/OCaml-4.01.0-ictce-5.5.0.eb]: I come to realized that if ncurses is there, I bet `libreadline`  should also be. ditto for goolf. ldd will prove the case

Comment (boegel) [inline easybuild/easyconfigs/o/OCaml/OCaml-4.01.0-ictce-5.5.0.eb]: seems not:

```
-bash-4.1$ ldd /gpfs/scratch/projects/project_gpilot/vsc40023/software/OCaml/4.01.0-ictce-5.5.0/bin/* | grep ncurses
	libncurses.so.5 => /gpfs/scratch/projects/project_gpilot/vsc40023/software/ncurses/5.9-ictce-5.5.0/lib/libncurses.so.5 (0x00002aaaab152000)
	libncurses.so.5 => /gpfs/scratch/projects/project_gpilot/vsc40023/software/ncurses/5.9-ictce-5.5.0/lib/libncurses.so.5 (0x00002aaaab152000)
-bash-4.1$ ldd /gpfs/scratch/projects/project_gpilot/vsc40023/software/OCaml/4.01.0-ictce-5.5.0/bin/* | grep readline
```

**Comment by rjeschmi** [comment]
There's an interesting mix here. Are some of these actually extensions of OCaml?

I haven't got my test space working yet for this yet, but I'll try to build tomorrow

**Comment by wpoely86** [comment]
ok to test

File: easybuild/easyconfigs/b/batteries/batteries-2.2-ictce-5.5.0-OCaml-4.01.0.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name = 'batteries'
```
Comment (wpoely86) [inline easybuild/easyconfigs/b/batteries/batteries-2.2-ictce-5.5.0-OCaml-4.01.0.eb]: this needs: `easyblock = 'OCamlPackage'` ?

**Comment by wpoely86** [comment]
@boegel all these easyconfig need a `easyblock` line

**Comment by boegel** [comment]
@wpoely86: actually, most of the easyconfigs should be removed here, since they can be included as extensions in the `OCaml` easyconfig instead (which is why this is still marked as WIP)

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6f7e12fd2660e1759dcb for a full test report.

File: easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+easyblock = 'MakeCp'
```
Comment (boegel) [inline easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb]: needs to be changed to `OCamlPackage`

Comment (boegel) [inline easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb]: well, maybe not... the build procedure here is quite different

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/987fb8bb7d46a9391501 for a full test report.

**Comment by boegel** [comment]
I think this is good to go (finally!).

Final review, anyone? @wpoely86?

File: easybuild/easyconfigs/o/OCaml/OCaml-4.01.0-goolf-1.4.10.eb [hunk 4]
```diff
@@ -18,24 +15,34 @@ description = """OCaml is a general purpose industrial-strength programming lang
...
-buildopts = "world.opt"
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+opam_ver = '1.2.2'
+source_urls = [
+    'http://caml.inria.fr/pub/distrib/ocaml-%s' % '.'.join(version.split('.')[:2]),
+    'https://github.com/ocaml/opam/releases/download/%s' % opam_ver,
+]
+sources = [
+    'ocaml-%(version)s.tar.gz',
```
Comment (wpoely86) [inline easybuild/easyconfigs/o/OCaml/OCaml-4.01.0-goolf-1.4.10.eb]: `NAMELOWER_TAR_GZ`?

Comment (boegel) [inline easybuild/easyconfigs/o/OCaml/OCaml-4.01.0-goolf-1.4.10.eb]: no, but `SOURCELOWER_TAR_GZ` would work ;)

File: easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+builddependencies = [
+    ('M4', '1.4.17'),
+]
+
+# see http://matsen.github.io/pplacer/compiling.html#opam
+prebuildopts = "rm -rf %(installdir)s && export OPAMROOT=%(installdir)s && opam init && "
+prebuildopts += "opam repo add pplacer-deps %(builddir)s/pplacer-opam-repository*/ && "
+prebuildopts += "opam update pplacer-deps && eval `opam config env` && "
+prebuildopts += "export OCAML_BACKEND=gcc && cat opam-requirements.txt | xargs opam install -y && "
+prebuildopts += "echo 'let version = \"v%(version)s\"\n' > common_src/version.ml && "
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb]: this hurts my eyes

Comment (boegel) [inline easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb]: meaning?

Comment (wpoely86) [inline easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb]: it needs a block?

Comment (boegel) [inline easybuild/easyconfigs/p/pplacer/pplacer-1.1.alpha16-ictce-5.5.0-OCaml-4.01.0.eb]: well, ok then...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/5e1ef8411e2473b56a1b for a full test report.

**Comment by boegel** [comment]
Tested on top of https://github.com/hpcugent/easybuild-easyblocks/pull/835, thanks for the feedback everyone!

**Comment by boegel** [comment]
Jenkins: test this please

**Pull request merged**